packages/office-ui-fabric-react/src/components/Slider/Slider.base.tsx

```diff
@@ -0,0 +1,306 @@
+import * as React from 'react';
+import {
+  BaseComponent,
+  KeyCodes,
+  css,
+  getId,
+  getRTL,
+  getRTLSafeKeyCode,
+  createRef
+} from '../../Utilities';
+import { ISliderProps, ISlider } from './Slider.types';
+import { Label } from '../../Label';
+import * as stylesImport from './Slider.scss';
+const styles: any = stylesImport;
+
+export interface ISliderState {
+  value?: number;
+  renderedValue?: number;
+}
+
+export enum ValuePosition {
+  Previous = 0,
+  Next = 1
+}
+
+export class SliderBase extends BaseComponent<ISliderProps, ISliderState> implements ISlider {
+  public static defaultProps: {} = {
+    step: 1,
+    min: 0,
+    max: 10,
+    showValue: true,
+    disabled: false,
+    vertical: false,
+    buttonProps: {}
+  };
+
+  private _sliderLine = createRef<HTMLDivElement>();
+  private _thumb = createRef<HTMLSpanElement>();
+  private _id: string;
+
+  constructor(props: ISliderProps) {
+    super(props);
+
+    this._warnMutuallyExclusive({
+      'value': 'defaultValue'
+    });
+
+    this._id = getId('Slider');
+
+    const value = props.value || props.defaultValue || props.min;
+
+    this.state = {
+      value: value,
+      renderedValue: value
+    };
+  }
+
+  /**
+   * Invoked when a component is receiving new props. This method is not called for the initial render.
+   */
+  public componentWillReceiveProps(newProps: ISliderProps): void {
+
+    if (newProps.value !== undefined) {
+      const value = Math.max(newProps.min as number, Math.min(newProps.max as number, newProps.value));
+
+      this.setState({
+        value: value,
+        renderedValue: value
+      });
+    }
+  }
+
+  public render(): React.ReactElement<{}> {
+    const {
+      ariaLabel,
+      className,
+      disabled,
+      label,
+      max,
+      min,
+      showValue,
+      buttonProps,
+      vertical
+    } = this.props;
+    const { value, renderedValue } = this.state;
+    const thumbOffsetPercent: number = (renderedValue! - min!) / (max! - min!) * 100;
+    const lengthString = vertical ? 'height' : 'width';
+    const onMouseDownProp: {} = disabled ? {} : { onMouseDown: this._onMouseDownOrTouchStart };
+    const onTouchStartProp: {} = disabled ? {} : { onTouchStart: this._onMouseDownOrTouchStart };
+    const onKeyDownProp: {} = disabled ? {} : { onKeyDown: this._onKeyDown };
+
+    return (
+      <div
+        className={ css('ms-Slider', styles.root, className, {
```

Comment:
For next PR: Make sure all this logic is in the styles file, including the globalClassNames such as "ms-Slider-enabled" etc. There shouldn't be a need for the css() function anywhere.